frontend/javascripts/dashboard/explorative_annotations_view.js

```diff
@@ -96,15 +97,21 @@ class ExplorativeAnnotationsView extends React.PureComponent<Props, State> {
     searchQuery: "",
     tags: [],
     isLoading: false,
+    performedInitialFetch: false,
   };
 
   componentDidMount() {
     this.setState(persistence.load(this.props.history));
-    this.fetchNextPage(0);
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(_prevProps, prevState) {
     persistence.persist(this.props.history, this.state);
+    if (
+      this.state.shouldShowArchivedTracings !== prevState.shouldShowArchivedTracings ||
+      !this.state.performedInitialFetch
+    ) {
+      this.fetchNextPage(0);
+    }
```

Comment:
Hmm, it's still a bit unusual, I guess, since `componentDidUpdate` is not called on the initial render (that's what `componentDidMount` is for). So, you are relying on the component updating, anyway, so that the fetch can be done. Probably this happens most of the time because of `this.setState(persistence.load(this.props.history));`, but it's not really safe. For example, if the localStorage is empty, the persistence module probably doesn't load, anything, so the state doesn't change?

I think, I would prefer to use this in `componentDidMount`:

```js
componentDidMount() {
    this.setState(persistence.load(this.props.history), () => this.fetchNextPage(0));
}
```

The callback is triggered after `setState` is done. Then, you also don't need `performedInitialFetch` if I'm not mistaken.

Reply:
> Probably this happens most of the time because of this.setState(persistence.load(this.props.history));, but it's not really safe.

Oh yeah, you are totally right here 👍. That's a nice suggestion. Thanks 🚀.

> The callback is triggered after setState is done. Then, you also don't need performedInitialFetch if I'm not mistaken.

👍